src/components/shared/Slider/Rail.tsx

```diff
@@ -1,15 +1,549 @@
-import React, { FC } from 'react';
-import styled from 'styled-components/native';
+import React, { FC, useEffect, useMemo } from 'react';
+import { StyleProp, StyleSheet, TouchableWithoutFeedback, ViewStyle } from 'react-native';
+import styled, { DefaultTheme, ThemeProps } from 'styled-components/native';
+
+interface RailThemeType {
+  backgroundColor: string;
+}
+
+interface MarkThemeType {
+  backgroundColor: string;
+}
+
+interface ThemeType extends DefaultTheme {
+  backgroundColor: string;
+  markColor: string;
+}
+
+interface StatefulThemeType extends ThemeType {
+  INVERTED: ThemeType;
+  DISABLED: ThemeType;
+}
+
+interface RailType extends ThemeProps<RailThemeType> {
+  theme: RailThemeType;
+}
+
+interface MarkType extends ThemeProps<MarkThemeType>{
+  theme: MarkThemeType;
+}
+
+interface MarkPositionerType {
+  width: number;
+  position: number;
+}
+
+interface Props {
+  testID?: string;
+  style?: StyleProp<ViewStyle>;
+  markStyle?: StyleProp<ViewStyle>;
+  mark?: React.ReactElement;
+  customMarkWidth?: number;
+  hideMark?: boolean;
+  step?: number;
+  pixelsPerStep?: number;
+  markCount?: number;
+  startMark?: boolean;
+  endMark?: boolean;
+  dark?: boolean;
+  lightblue?: boolean;
+  blue?: boolean;
```

Comment:
What do you think about making a prop like `variant` or `theme`?
You can give it a type of theme. 
For example `variant=dark` or `variant=blue` or `variant=lightblue`.
Because the user can't know which one is applied if `dark=true` and `blue=true`.

Reply:
Yeah, as you mentioned, users can't know how it works.

`variant` is a good idea and I considered using that kind of prop. But the prop has a drawback that users have to hard-code like "blue", "dark"... That's why I didn't use that.

However, **thanks to your description, `variant` looks much better even for the future. I'll fix it! :)**

I didn't want to export a constant object for values like VARIANT.BLUE. I think it is a little dirty that users have to take more things to use the prop.
But I should code the constant because I don't see a better idea to avoid hard-coding. :(
(It probably not a serious thing, because it is just a library supporting API doc README.md. 👍 )

p.s.
I tried to use `theme` prop before when I implement other component. There are some problems.
The problems are a little complicated to describe here.
A discussion for `theme` prop: https://github.com/dooboolab/dooboo-ui-native/pull/28#issuecomment-553315084

@tony3918
Could you check this review to write the test code as well? :)